image_segmentation/pytorch/runtime/distributed/distributed_utils.py

```diff
@@ -0,0 +1,186 @@
+import random
+from typing import List, Tuple
+
+import numpy as np
+import torch
+from runtime.distributed.cuda_strategy import CUDAStrategy
+from runtime.distributed.distributed_strategy import DistributedStrategy
+from runtime.distributed.xla_strategy import XLAStrategy
+
+_STRATEGY: DistributedStrategy = None
+
+
+def init_distributed(flags) -> bool:
+    """
+    Initializes the distributed backends (either PyTorch CUDA or PyTorch/XLA)
+
+    :param flags: the runtime arguments
+    :return: false if using single core training, true otherwise
+    :rtype: bool
+    """
+    global _STRATEGY
+    if flags.torch_xla:
+        _STRATEGY = XLAStrategy()
+    else:
+        _STRATEGY = CUDAStrategy()
+    return get_world_size() > 1
+
+
+def _assert_initialized(func):
+    def wrapper(*args, **kwargs):
```

Comment:
Consider using `functools.wraps` here

Explanation of what it is and why it's useful: https://stackoverflow.com/a/309000/13708606

Reply:
That's a great suggestion. I didn't know about `functools.wraps`. I have added it to my decorator now.